components/cloudflare_api_key/cloudflare_api_key.app.mjs

```diff
@@ -1,11 +1,331 @@
+      type: "string",
+      label: "Zone ID",
+      description: "The zone identifier",
+      async options({ prevContext }) {
+        const page = prevContext.page || 1;
+        const zones = await this.getZones({
+          page: page,
+        });
+
+        return {
+          options: zones.result.map((zone) => ({
+            value: zone.id,
+            label: zone.name,
+          })),
+          context: {
+            page: page + 1,
+          },
+        };
+      },
+    },
+    dnsRecordIdentifier: {
+      type: "string",
+      label: "DNS record ID",
+      description: "The DNS record identifier",
+      async options({
+        prevContext,
+        zoneIdentifier,
+      }) {
+        const page = prevContext.page || 1;
+        const dnsRecords = await this.listDnsRecords(zoneIdentifier, {
+          page: page,
+        });
+
+        return {
+          options: dnsRecords.result.map((record) => ({
+            value: record.id,
+            label: record.name,
+          })),
+          context: {
+            page: page + 1,
+          },
+        };
+      },
+    },
+    certificateIdentifier: {
+      type: "string",
+      label: "Certificate ID",
+      description: "The certificate identifier",
+      async options({
+        prevContext,
+        zoneIdentifier,
+      }) {
+        const page = prevContext.page || 1;
+        const certificates = await this.getCertificates(zoneIdentifier, {
+          page: page,
+        });
+
+        return {
+          options: certificates.result.map((certificate) => certificate.id),
+          context: {
+            page: page + 1,
+          },
+        };
+      },
+    },
+    dnsRecordType: {
+      type: "string",
+      label: "Type",
+      description: "DNS record type",
+      options() {
+        return [
+          "A",
+          "AAAA",
+          "CNAME",
+          "HTTPS",
+          "TXT",
+          "SRV",
+          "LOC",
+          "MX",
+          "NS",
+          "CERT",
+          "DNSKEY",
+          "DS",
+          "NAPTR",
+          "SMIMEA",
+          "SSHFP",
+          "SVCB",
+          "TLSA",
+          "URI read only",
+        ];
+      },
+    },
+    dnsName: {
+      type: "string",
+      label: "Name",
+      description: "DNS record name",
+    },
+    dnsRecordContent: {
+      type: "string",
+      label: "Content",
+      description: "DNS record content",
+    },
+    dnsRecordProxied: {
+      type: "boolean",
+      label: "Proxied",
+      description: "Whether the record is receiving the performance and security benefits of Cloudflare",
+    },
+    dnsRecordTtl: {
+      type: "integer",
+      label: "TTL",
+      description: "Time to live, in seconds, of the DNS record. Must be between 60 and 86400, or 1 for 'automatic'",
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _makeRequest($ = this, opts) {
```

Comment:
Should this method be `async`?

Reply:
it shouldn't, cause `wait` feature its not being used in this method